### ISSUE TITLE
How to use LightGBM to fit a function curve?

### ISSUE THREAD

**Issue by yonghaoXuGit** [lightgbm-org/LightGBM#1917]
I want to use LightGBM to fit a function curve,but in the examples of LightGBM's dataset,every record has a label column. I don't know how to create my training dataset and testset.

**Comment by chivee** [comment]
if the function you want to learn can be formalized as `f(x) = y` and the label should `y`.

**Comment by guolinke** [comment]
@yonghaoXuGit 
you can try following parameters:
```
objective = regression
num_trees = 500
learning_rate = 0.05
num_leaves = 100
reg_sqrt=False
boost_from_average=False
```

**Comment by yonghaoXuGit** [comment]
Thanks，that's my fault，it should be split by tab not blank.

**Comment by yonghaoXuGit** [comment]
And I have another question:are there any other restrictions on predicted data?
Because I found that the prediction result is dependent on the data's property severely.
